io.openems.edge.evcs.webasto.unite/src/io/openems/edge/evcs/webasto/unite/WebastoImpl.java

```diff
@@ -0,0 +1,304 @@
+	 * Handles charge states.
+	 */
+	private final ChargeStateHandler chargeStateHandler = new ChargeStateHandler(this);
+
+	/**
+	 * Processes the controller's writes to this evcs component.
+	 */
+	private final WriteHandler writeHandler = new WriteHandler(this);
+
+	public WebastoImpl() {
+		super(//
+				OpenemsComponent.ChannelId.values(), //
+				Webasto.ChannelId.values(), //
+				Evcs.ChannelId.values(), //
+				ModbusComponent.ChannelId.values(), //
+				ManagedEvcs.ChannelId.values() //
+		);
+	}
+
+	@Activate
+	void activate(ComponentContext context, Config config) throws ConfigurationException, OpenemsException {
+		if (super.activate(context, config.id(), config.alias(), config.enabled(), config.modbusUnitId(), this.cm,
+				"Modbus", config.modbus_id())) {
+			return;
+		}
+		this.config = config;
+		this._setChargingType(ChargingType.AC);
+		this._setPowerPrecision(230); // 1A steps
+		this._setFixedMinimumHardwarePower(this.getConfiguredMinimumHardwarePower());
+		this._setFixedMaximumHardwarePower(this.getConfiguredMaximumHardwarePower());
+		this.readHandler = new WebastoReadHandler(this);
+
+		/*
+		 * Calculates the maximum and minimum hardware power dynamically by listening on
+		 * the fixed hardware limit and the phases used for charging
+		 */
+		Evcs.addCalculatePowerLimitListeners(this);
+
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		super.deactivate();
+	}
+
+	@Override
+	@Reference(policy = ReferencePolicy.STATIC, policyOption = ReferencePolicyOption.GREEDY, cardinality = ReferenceCardinality.MANDATORY)
+	protected void setModbus(BridgeModbus modbus) {
+		super.setModbus(modbus);
+	}
+
+	@Override
+	protected ModbusProtocol defineModbusProtocol() throws OpenemsException {
+		var modbusProtocol = new ModbusProtocol(this,
+				new FC4ReadInputRegistersTask(100, Priority.LOW,
+						m(Webasto.ChannelId.SERIAL_NUMBER, new StringWordElement(100, 25))),
+				new FC4ReadInputRegistersTask(130, Priority.LOW,
+						m(Webasto.ChannelId.CHARGE_POINT_ID, new StringWordElement(130, 50))),
+				new FC4ReadInputRegistersTask(190, Priority.LOW,
+						m(Webasto.ChannelId.BRAND, new StringWordElement(190, 10))),
+				new FC4ReadInputRegistersTask(210, Priority.LOW,
+						m(Webasto.ChannelId.MODEL, new StringWordElement(210, 5))),
+				new FC4ReadInputRegistersTask(230, Priority.LOW,
+						m(Webasto.ChannelId.FIRMWARE_VERSION, new StringWordElement(230, 50))),
+				new FC4ReadInputRegistersTask(290, Priority.LOW,
+						m(Webasto.ChannelId.DATE, new UnsignedDoublewordElement(290))),
+				new FC4ReadInputRegistersTask(294, Priority.LOW,
+						m(Webasto.ChannelId.TIME, new UnsignedDoublewordElement(294))),
+				new FC4ReadInputRegistersTask(400, Priority.LOW,
+						m(Webasto.ChannelId.CHARGE_POINT_POWER, new UnsignedDoublewordElement(400))),
+				new FC4ReadInputRegistersTask(404, Priority.LOW,
+						m(Webasto.ChannelId.NUMBER_OF_PHASES, new UnsignedWordElement(404))),
+				new FC4ReadInputRegistersTask(1000, Priority.LOW,
```

Comment:
Ideally these FC4ReadInputRegistersTasks should be joint. Do I remember correctly, that the Webasto did not support reading multiple registers with 'gaps' inbetween? Then we should possible add a small comment, as this is very untypical. (i.e. people should not take this approach as 'best-practice')